gcp/src/main/java/org/apache/iceberg/gcp/biglake/BigLakeCatalog.java

```diff
@@ -0,0 +1,405 @@
+  private FileSystem fs;
+  private FileIO fileIO;
+  private Configuration conf;
+  private String projectId;
+  private String location;
+  // BLMS catalog ID and fully qualified name.
+  private String catalogId;
+  private CatalogName catalogName;
+  private BigLakeClient client;
+
+  // Must have a no-arg constructor to be dynamically loaded
+  // initialize(String name, Map<String, String> properties) will be called to complete
+  // initialization
+  public BigLakeCatalog() {}
+
+  @Override
+  public void initialize(String inputName, Map<String, String> properties) {
+    if (!properties.containsKey(PROPERTIES_KEY_GCP_PROJECT)) {
+      throw new ValidationException("GCP project must be specified");
+    }
+    String propProjectId = properties.get(PROPERTIES_KEY_GCP_PROJECT);
+    String propLocation =
+        properties.getOrDefault(PROPERTIES_KEY_GCP_LOCATION, DEFAULT_GCP_LOCATION);
+    BigLakeClient newClient;
+    try {
+      newClient =
+          new BigLakeClientImpl(
+              properties.getOrDefault(
+                  PROPERTIES_KEY_BIGLAKE_ENDPOINT, DEFAULT_BIGLAKE_SERVICE_ENDPOINT),
+              propProjectId,
+              propLocation);
+    } catch (IOException e) {
+      throw new ServiceFailureException(e, "Creating BigLake client failed");
+    }
+    initialize(inputName, properties, propProjectId, propLocation, newClient);
+  }
+
+  @VisibleForTesting
+  void initialize(
+      String inputName,
+      Map<String, String> properties,
+      String propProjectId,
+      String propLocation,
+      BigLakeClient bigLakeClient) {
+    this.catalogPulginName = inputName;
+    this.catalogProperties = ImmutableMap.copyOf(properties);
+    this.projectId = propProjectId;
+    this.location = propLocation;
+    Preconditions.checkNotNull(bigLakeClient, "BigLake client must not be null");
+    this.client = bigLakeClient;
+
+    if (this.conf == null) {
+      LOG.warn("No Hadoop Configuration was set, using the default environment Configuration");
+      this.conf = new Configuration();
+    }
+
+    // Users can specify the BigLake catalog ID, otherwise catalog plugin will be used.
+    this.catalogId = properties.getOrDefault(PROPERTIES_KEY_BLMS_CATALOG, inputName);
+    this.catalogName = CatalogName.of(projectId, location, catalogId);
+    LOG.info("Use BigLake catalog: {}", catalogName.toString());
+
+    if (properties.containsKey(CatalogProperties.WAREHOUSE_LOCATION)) {
+      this.conf.set(
+          HIVE_METASTORE_WAREHOUSE_DIR,
+          LocationUtil.stripTrailingSlash(properties.get(CatalogProperties.WAREHOUSE_LOCATION)));
+    }
+
+    this.fs =
+        Util.getFs(
+            new Path(
+                LocationUtil.stripTrailingSlash(
+                    properties.get(CatalogProperties.WAREHOUSE_LOCATION))),
+            conf);
+
+    String fileIOImpl =
+        properties.getOrDefault(
+            CatalogProperties.FILE_IO_IMPL, "org.apache.iceberg.hadoop.HadoopFileIO");
```

Comment:
`ResolvingFileIO` should be used as the default.

Reply:
Updated to use ResolvingFileIO.